webUI/react/src/components/reader/pager/VerticalPager.tsx

```diff
@@ -29,36 +29,87 @@ export default function VerticalReader(props: IReaderProps) {
     const classes = useStyles();
     const history = useHistory();
 
-    const [initialScroll, setInitialScroll] = useState(-1);
-    const initialPageRef = useRef<HTMLDivElement>(null);
+    const pageRef = useRef<HTMLDivElement>(null);
+    const pagesRef = useRef<HTMLDivElement[]>([]);
+    let readerCurPage = curPage;
```

Comment:
see if you can write this with only const variables?

Generally in React Functional components you should have everything as const(following functional programming rules) and not do things like this. specially you have to give extra attention when dealing with refs.